ksql-rest-app/src/test/java/io/confluent/ksql/rest/integration/ClusterTerminationTest.java

```diff
@@ -0,0 +1,207 @@
+import io.confluent.ksql.rest.entity.CommandStatusEntity;
+import io.confluent.ksql.rest.entity.KsqlEntity;
+import io.confluent.ksql.rest.entity.KsqlEntityList;
+import io.confluent.ksql.rest.entity.KsqlRequest;
+import io.confluent.ksql.rest.server.TestKsqlRestApp;
+import io.confluent.ksql.util.KsqlConstants;
+import io.confluent.ksql.util.PageViewDataProvider;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.RuleChain;
+
+@SuppressWarnings("unchecked")
+@Category({IntegrationTest.class})
+public class ClusterTerminationTest {
+
+  private static final String PAGE_VIEW_TOPIC = "pageviews";
+  private static final String PAGE_VIEW_STREAM = "pageviews_original";
+  private static final PageViewDataProvider PAGE_VIEW_DATA_PROVIDER = new PageViewDataProvider();
+
+  private static final String SINK_TOPIC = "sink_topic";
+  private static final String SINK_STREAM = "sink_stream";
+
+  private static final IntegrationTestHarness TEST_HARNESS = IntegrationTestHarness.build();
+
+  private static final TestKsqlRestApp REST_APP = TestKsqlRestApp
+      .builder(TEST_HARNESS::kafkaBootstrapServers)
+      .buildWithServiceContext(TEST_HARNESS::serviceContext);
+
+  @ClassRule
+  public static final RuleChain CHAIN = RuleChain.outerRule(TEST_HARNESS).around(REST_APP);
+
+  private KsqlRestClient restClient;
+
+  @BeforeClass
+  public static void setUpClass() {
+    TEST_HARNESS.ensureTopics(PAGE_VIEW_TOPIC);
+
+    createStreams(PAGE_VIEW_STREAM);
+  }
+
+  @Before
+  public void setUp() {
+    restClient = REST_APP.buildKsqlClient();
+  }
+
+  @After
+  public void cleanUp() {
+    restClient.close();
+  }
+
+  @Test
+  public void shouldCleanUpSinkTopicsAndSchemasDuringClusterTermination() throws Exception {
+    // Given:
+    makeKsqlRequest("CREATE STREAM " + SINK_STREAM
+        + " WITH (kafka_topic='" + SINK_TOPIC + "',value_format='avro')"
+        + " AS SELECT * FROM " + PAGE_VIEW_STREAM + ";"
+    );
+
+    TEST_HARNESS.verifyTopicsPresent(SINK_TOPIC);
+
+    // Produce to stream so that schema is registered by AvroConverter
+    TEST_HARNESS.produceRows(PAGE_VIEW_TOPIC, PAGE_VIEW_DATA_PROVIDER, JSON, System::currentTimeMillis);
+
+    TEST_HARNESS.verifySubjectPresent(SINK_TOPIC + KsqlConstants.SCHEMA_REGISTRY_VALUE_SUFFIX);
+
+    // When:
+    terminateCluster(ImmutableList.of(SINK_TOPIC));
+
+    // Then:
+    TEST_HARNESS.verifyTopicsAbsent(SINK_TOPIC);
+
+    TEST_HARNESS.verifySubjectAbsent(SINK_TOPIC + KsqlConstants.SCHEMA_REGISTRY_VALUE_SUFFIX);
+  }
+
+  private static List<KsqlEntity> awaitResults(final List<KsqlEntity> pending) {
```

Comment:
This helper method and a bunch of the others below are duplicated here, and in RestApiTest and KsqlResourceFunctionalTest. I think they should be pulled out into a shared test util file, but would prefer making this change in a follow-up PR unless others feel strongly.